From 12e6a121b85456b8c50e2dff3b87032a6d61e773 Mon Sep 17 00:00:00 2001
From: Vaishnav M A <vaishnav@beagleboard.org>
Date: Wed, 30 Dec 2020 18:54:37 +0530
Subject: [PATCH 08/41] w1: change device name in w1_attach_slave

currently if there is more than one device with the same
ROM ID, there will be a conflict even if the devices are
on different w1 master.

Signed-off-by: Vaishnav M A <vaishnav@beagleboard.org>
---
 drivers/w1/w1.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/w1/w1.c b/drivers/w1/w1.c
index 15a2ee32f116..e4593174973a 100644
--- a/drivers/w1/w1.c
+++ b/drivers/w1/w1.c
@@ -681,7 +681,8 @@ static int __w1_attach_slave_device(struct w1_slave *sl)
 	sl->dev.of_node = of_find_matching_node(sl->master->dev.of_node,
 						sl->family->of_match_table);
 
-	dev_set_name(&sl->dev, "%02x-%012llx",
+	dev_set_name(&sl->dev, "%s-%02x-%012llx",
+		 sl->master->name,
 		 (unsigned int) sl->reg_num.family,
 		 (unsigned long long) sl->reg_num.id);
 	snprintf(&sl->name[0], sizeof(sl->name),
-- 
2.30.2

